source/Octopus.Tentacle/Kubernetes/KubernetesPodMonitor.cs

```diff
@@ -69,27 +75,35 @@ async Task UpdateLoop(CancellationToken cancellationToken)
         internal async Task<string> InitialLoadAsync(CancellationToken cancellationToken)
         {
             log.Verbose("Preloading pod statuses");
-            //clear the status'
-            podStatusLookup.Clear();
 
+            var newStatuses = new ConcurrentDictionary<ScriptTicket, PodStatus>();
             var allPods = await podService.ListAllPodsAsync(cancellationToken);
             foreach (var pod in allPods.Items)
             {
-                var status = new PodStatus(pod.GetScriptTicket());
+                var scriptTicket = pod.GetScriptTicket();
+                if (!podStatusLookup.TryGetValue(scriptTicket, out var status))
+                {
+                    status = new PodStatus(scriptTicket, clock);
+                }
                 status.Update(pod);
 
                 log.Verbose($"Preloaded pod {pod.Name()}. {status}");
-                podStatusLookup[status.ScriptTicket] = status;
+                newStatuses[scriptTicket] = status;
             }
 
+            // Updating a reference is an atomic operation
+            // and we only add data to this dictionary
+            // in this class so this is thread safe.
+            podStatusLookup = newStatuses;
```

Comment:
@APErebus I've updated the dictionary to be a `ConcurrentDictionary<>` and the main difference is that now we load up a new instance of the dictionary and just set the reference on the field. This is an atomic operation as far as I'm aware and because we don't do any updates to the dictionary outside of this class, we can be sure that no new values are added to the old podStatusLookup while we are building the new instance.

Reply:
yeah, that's a good idea 👍